tests/network/vmi_istio.go

```diff
@@ -0,0 +1,244 @@
+	vmi := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskCirros))
+	tests.AddServiceAccountDisk(vmi, serviceAccount)
+	tests.AddUserData(vmi, "disk1", userData)
+	addInjectIstioProxyAnnotation(vmi, istioProxyAnnotationValue)
+	addAppLabel(vmi, appLabel)
+	vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*iface}
+	vmi.Spec.Networks = []v1.Network{*network}
+	return vmi
+}
+
+var _ = FSIGDescribe("[Serial] Istio", func() {
+	var err error
+	var virtClient kubecli.KubevirtClient
+	var vmiWithExplicitPorts *v1.VirtualMachineInstance
+	var vmiWithDefaultPorts *v1.VirtualMachineInstance
+
+	defaultNetwork := v1.Network{
+		Name: "default",
+		NetworkSource: v1.NetworkSource{
+			Pod: &v1.PodNetwork{},
+		},
+	}
+	defaultInterface := v1.Interface{
+		Name: "default",
+		InterfaceBindingMethod: v1.InterfaceBindingMethod{
+			Masquerade: &v1.InterfaceMasquerade{},
+		},
+	}
+	interfaceWithExplicitPorts := v1.Interface{
+		Name: "default",
+		InterfaceBindingMethod: v1.InterfaceBindingMethod{
+			Masquerade: &v1.InterfaceMasquerade{},
+		},
+		Ports: []v1.Port{
+			{
+				Port: svcDeclaredTestPort,
+			},
+			{
+				Port: svcUndeclaredTestPort,
+			},
+		},
+	}
+
+	Describe("Virtual machine in a service mesh using masquerade interface with explicit port list", func() {
+		tests.BeforeAll(func() {
+			if !istioServiceMeshDeployed() {
+				Skip("Istio service mesh is required for service-mesh tests to run")
+			}
+
+			tests.BeforeTestCleanup()
+			virtClient, err = kubecli.GetKubevirtClient()
+			tests.PanicOnError(err)
+
+			vmiWithExplicitPorts = newRandomCirrosVMIWithIstioProxyAndEphemeralDiskAndUserDataAndAndServiceAccountDiskAndNetworkAndInterface(
+				"default",
+				vmiExplicitPortsAppSelector,
+				"true",
+				"#!/bin/bash\necho 'hello'\n",
+				&defaultNetwork,
+				&interfaceWithExplicitPorts,
+			)
+			vmiWithDefaultPorts = newRandomCirrosVMIWithIstioProxyAndEphemeralDiskAndUserDataAndAndServiceAccountDiskAndNetworkAndInterface(
+				"default",
+				vmiDefaultPortsAppSelector,
+				"true",
+				"#!/bin/bash\necho 'hello'\n",
+				&defaultNetwork,
+				&defaultInterface,
+			)
+			vmiWithExplicitPortsService := newServiceWithAppSelector(fmt.Sprintf("%s-service", vmiWithExplicitPorts.Name), vmiWithExplicitPorts.Namespace, vmiExplicitPortsAppSelector)
+			vmiWithDefaultPortsService := newServiceWithAppSelector(fmt.Sprintf("%s-service", vmiWithDefaultPorts.Name), vmiWithDefaultPorts.Namespace, vmiDefaultPortsAppSelector)
+
+			addLabelToNamespace(vmiWithExplicitPorts.Namespace, istioInjectNamespaceLabel, "enabled")
+			for _, svc := range []corev1.Service{*vmiWithExplicitPortsService, *vmiWithDefaultPortsService} {
+				_, err = virtClient.CoreV1().Services(svc.Namespace).Create(context.Background(), &svc, metav1.CreateOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
+			}
+			for _, networkVMI := range []*v1.VirtualMachineInstance{vmiWithExplicitPorts, vmiWithDefaultPorts} {
```

Comment:
Maybe merge this for-loop and the next one into on loop?

Reply:
Done